mypy.ini

```
@@ -1,17 +1,24 @@
 [mypy]
 files =
     superduperdb/apis/retry.py,
-    superduperdb/datalayer/base/artifacts.py,
-    superduperdb/datalayer/base/imports.py,
+    superduperdb/datalayer/**/*.py,
     superduperdb/misc/config.py,
     superduperdb/server/*.py,
     superduperdb/vector_search/base.py,
     superduperdb/vector_search/inmemory.py,
     superduperdb/vector_search/milvus.py,
     tests/integration/**/*.py
 
-[mypy-pymilvus.*]
+[mypy-dill.*]
 ignore_missing_imports = True
 
-[mypy-dill.*]
+[mypy-networkx.*]
 ignore_missing_imports = True
+
+[mypy-pymilvus.*]
+ignore_missing_imports = True
+
+[mypy-superduperdb.datalayer.mongodb.database]
+disable_error_code =
```

So what's this mean?  Isn't there a way around it without suppressing these errors?

---

This is related to the Note I've left:

> It has to be noted without this we get the following error message:
> 
> ```shell
> superduperdb/datalayer/mongodb/database.py:50: error: Signature of "_base_delete" incompatible with supertype "BaseDatabase"  [override]
> superduperdb/datalayer/mongodb/database.py:50: note:      Superclass:
> superduperdb/datalayer/mongodb/database.py:50: note:          def _base_delete(self, delete: Delete) -> Any
> superduperdb/datalayer/mongodb/database.py:50: note:      Subclass:
> superduperdb/datalayer/mongodb/database.py:50: note:          def _base_delete(self, delete: Delete, **kwargs: Any) -> Any
> superduperdb/datalayer/mongodb/database.py:53: error: Argument 1 of "_base_insert" is incompatible with supertype "BaseDatabase"; supertype defines the argument type as "Insert"  [override]
> superduperdb/datalayer/mongodb/database.py:53: note: This violates the Liskov substitution principle
> superduperdb/datalayer/mongodb/database.py:53: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
> superduperdb/datalayer/mongodb/database.py:56: error: Argument 1 of "_base_update" is incompatible with supertype "BaseDatabase"; supertype defines the argument type as "Update"  [override]
> superduperdb/datalayer/mongodb/database.py:56: note: This violates the Liskov substitution principle
> superduperdb/datalayer/mongodb/database.py:56: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
> superduperdb/datalayer/mongodb/database.py:73: error: Argument 1 of "_get_cursor" is incompatible with supertype "BaseDatabase"; supertype defines the argument type as "Select"  [override]
> superduperdb/datalayer/mongodb/database.py:73: note: This violates the Liskov substitution principle
> superduperdb/datalayer/mongodb/database.py:73: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
> superduperdb/datalayer/mongodb/database.py:86: error: Argument 1 of "_get_ids_from_select" is incompatible with supertype "BaseDatabase"; supertype defines the argument type as "Select"  [override]
> superduperdb/datalayer/mongodb/database.py:86: note: This violates the Liskov substitution principle
> superduperdb/datalayer/mongodb/database.py:86: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
> superduperdb/datalayer/mongodb/database.py:92: error: Argument 1 of "_get_raw_cursor" is incompatible with supertype "BaseDatabase"; supertype defines the argument type as "Select"  [override]
> superduperdb/datalayer/mongodb/database.py:92: note: This violates the Liskov substitution principle
> superduperdb/datalayer/mongodb/database.py:92: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
> Found 6 errors in 1 file (checked 27 source files)
> ```
> The code can be changed in `superduperdb/datalayer/mongodb/database.py` but it would be bodged solution that would complicate the codebase unnecessarily, both with a more generic definition signature and additional asserts.
> 
> It is also important to reconsider the current mehod overrides of `superduperdb.datalayer.base.database.BaseDatabase` as part of a new development. 

The other solution would involve changing the annotation of the functions to include the types of the Base Class and then adding an `assert` instruction to ensure that the type of the argument (related to this problem) is not of said Class.